qi-doc/scribblings/field-guide.scrbl

```diff
@@ -403,7 +403,7 @@ Yet, either implementation produces the same output: @racket[(list 1 9 25)].
 
 So, to reiterate, while the output of Qi flows will be the same as the output of equivalent Racket expressions, they may nevertheless exhibit a different order of effects.
 
-If you'd like to ensure a particular order of effects, use @racket[effect] at the appropriate points in your flow. If you'd like to use Racket's order of effects, define your flow using @racket[esc] (although this would lose any Qi compiler optimizations).
+If you'd like to ensure a particular order of effects, use @racket[effect] at the appropriate points in your flow. If you'd like to use Racket's order of effects, @seclink["Using_Racket_to_Define_Flows"]{define your flow in Racket} by using @racket[esc].
```

Comment:
I think this too is arising out of the confusion re: the term "define," and I'll see about avoiding this term. Both sentences here are just talking about use of the form rather than binding the resulting flow to an identifier. Based on the suggested formalization of effect guarantees in the other comment, I think only the claim about `esc` here is true, that is, if you use `esc` to wrap your flow specification, then everything inside will be left alone by the Qi compiler (unless you have a nested entry into Qi, such as a use of `flow`, of course!).

So re: `effect`, I think its use is likely to be more about _style_ rather than about theoretical guarantees. That is, you should separate your effects from the other computations because it's _cleaner_ to do it that way, and encourages use of pure functions which are more portable, and encourages smaller, more modular functions which are better building blocks. But I think `(effect e f)` is identical, as far as guarantees regarding effects, to simply `f'`, where `f'` is the same as `f` but also includes the effect `e` internally.

I still feel we should be encouraging separation of effects and keeping functions pure (i.e. in the other section, not here specifically), but for broader/different reasons (which maybe we should elaborate on) than order of effects.